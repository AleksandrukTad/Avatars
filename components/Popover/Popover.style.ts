import styled, { css, keyframes } from "styled-components";

const arrowHeight = 10;

const visibilityAnimation = keyframes`
  from {
    opacity: 0;
  } to {
    opacity: 1;
  }
`;

interface IWrapperProps {
  open: boolean;
}

export const Wrapper = styled.div<IWrapperProps>`
  display: none;
  flex-flow: column;
  align-items: center;
  padding: 1rem 0;
  background-color: ${({ theme }) => theme.Popover};
  top: calc(100% + ${arrowHeight}px);
  position: absolute;
  width: 100px;
  border-radius: 5px;
  z-index: 1;

  ${({ open }) =>
    open &&
    css`
      display: flex;
      animation: ${visibilityAnimation} 0.25s ease;
    `};

  &:before {
    position: absolute;
    content: "";
    width: 20px;
    height: 20px;
    /* -5 to hide rounded sides and make sure that it does nto touch the avatar*/
    top: -${arrowHeight - 5}px;
    left: 50%;
    background-color: ${({ theme }) => theme.Popover};
    border-radius: 5px;
    /* arrow is square div with the same color as bg, just rotated and rounded */
    transform: translate3d(-50%, 0, 0) rotate(45deg);
    /* hidden arrow below wrapper */
    z-index: -1;
  }
`;
