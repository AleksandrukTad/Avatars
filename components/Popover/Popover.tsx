import React from "react";

import { Wrapper } from "./Popover.style";

interface IProps extends React.Props<any> {
  open: boolean;
}

const Popover: React.FunctionComponent<IProps> = ({ open, children }) => {
  return <Wrapper open={open}> {children} </Wrapper>;
};

export default Popover;
