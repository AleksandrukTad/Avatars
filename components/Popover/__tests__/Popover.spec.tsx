import "jest-dom/extend-expect";
import "jest-styled-components";

// __tests__/fetch.js
import { cleanup, render } from "@testing-library/react";
import React from "react";
// this adds custom jest matchers from jest-dom

import Popover from "../Popover";

// automatically unmount and cleanup DOM after the test is finished.
afterEach(cleanup);

test("Render open popover", async () => {
  const { asFragment } = render(<Popover open={true} />);
  // you can also get a `DocumentFragment`, which is useful if you want to compare nodes across renders
  expect(asFragment()).toMatchSnapshot();
});

test("Render closed popover", async () => {
  const { asFragment } = render(<Popover open={false} />);
  // you can also get a `DocumentFragment`, which is useful if you want to compare nodes across renders
  expect(asFragment()).toMatchSnapshot();
});
