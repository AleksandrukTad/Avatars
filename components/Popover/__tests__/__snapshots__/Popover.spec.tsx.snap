// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render closed popover 1`] = `
<DocumentFragment>
  .c0 {
  display: none;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0;
  top: calc(100% + 10px);
  position: absolute;
  width: 100px;
  border-radius: 5px;
  z-index: 1;
}

.c0:before {
  position: absolute;
  content: "";
  width: 20px;
  height: 20px;
  top: -5px;
  left: 50%;
  border-radius: 5px;
  -webkit-transform: translate3d(-50%,0,0) rotate(45deg);
  -ms-transform: translate3d(-50%,0,0) rotate(45deg);
  transform: translate3d(-50%,0,0) rotate(45deg);
  z-index: -1;
}

<div
    class="c0"
  >
      
  </div>
</DocumentFragment>
`;

exports[`Render open popover 1`] = `
<DocumentFragment>
  .c0 {
  display: none;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0;
  top: calc(100% + 10px);
  position: absolute;
  width: 100px;
  border-radius: 5px;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-animation: eMLfYp 0.25s ease;
  animation: eMLfYp 0.25s ease;
}

.c0:before {
  position: absolute;
  content: "";
  width: 20px;
  height: 20px;
  top: -5px;
  left: 50%;
  border-radius: 5px;
  -webkit-transform: translate3d(-50%,0,0) rotate(45deg);
  -ms-transform: translate3d(-50%,0,0) rotate(45deg);
  transform: translate3d(-50%,0,0) rotate(45deg);
  z-index: -1;
}

<div
    class="c0"
    open=""
  >
      
  </div>
</DocumentFragment>
`;
